File: app/src/main/java/com/mehul/redditwall/favorites/FavDAO.java
package com.mehul.redditwall.favorites;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;

import java.util.List;

@Dao
public interface FavDAO {
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insert(FavImage saved);

    @Query("DELETE FROM fav_table")
    void deleteAll();

    @Query("SELECT * from fav_table ORDER BY fav_url ASC")
    LiveData<List<FavImage>> getAllFavImages();

    @Query("SELECT * from fav_table ORDER BY fav_url ASC")
    List<FavImage> getFavAsList();

    @Delete
    void deleteFavImage(FavImage saved);
}

File: app/src/main/java/com/mehul/redditwall/favorites/FavDAO.kt
package com.mehul.redditwall.favorites

import androidx.lifecycle.LiveData
import androidx.room.*

@Dao
interface FavDAO {
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insert(saved: FavImage?)

    @Query("DELETE FROM fav_table")
    fun deleteAll()

    @get:Query("SELECT * from fav_table ORDER BY fav_url ASC")
    val allFavImages: LiveData<List<FavImage?>?>?

    @get:Query("SELECT * from fav_table ORDER BY fav_url ASC")
    val favAsList: List<FavImage?>?

    @Delete
    fun deleteFavImage(saved: FavImage?)
}